onward/app/feed/MostViewed.scala

```diff
@@ -0,0 +1,41 @@
+package feed
+
+import com.gu.commercial.branding.{Branding, BrandingFinder}
+import common.{Edition, Logging}
+import contentapi.{ContentApiClient, QueryDefaults}
+import model.RelatedContentItem
+import services.MostReadItem
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+object MostViewed extends Logging {
+
+  def relatedContentItems(ophanMostViewed: Future[Seq[MostReadItem]], edition: Edition = Edition.defaultEdition)
+                         (contentApiClient: ContentApiClient)
+                         (implicit ec: ExecutionContext): Future[Seq[Option[RelatedContentItem]]] = {
+
+    ophanMostViewed.flatMap { allMostViewed =>
+      val allRelatedContentItems: Seq[Future[Option[RelatedContentItem]]] = allMostViewed.map { mostReadItem =>
+        val url = mostReadItem.url
+        contentApiClient
+          .getResponse(contentApiClient
+            .item(urlToContentPath(url), edition)
+            .showSection(true)
+            .showFields((QueryDefaults.trailFields :+ "isInappropriateForSponsorship").mkString(",")))
+          .map(
+            _.content
+            .filterNot { content => BrandingFinder.findBranding(edition.id)(content).exists(_.isPaid)}
```

Comment:
So this has now changed from filtering out paid content from not only the most read component, but also from all components that use agents to store things that are "related"?